### PR TITLE
Issue 606

### LINKED ISSUE #606: Table.collapse in one_to_many was not producing the right data shape
This issue wasn't previously identified as it is likely that only picrust uses this path through the code. The logic changed some when we merged the collapse methods from 1.3.1 to 2.x.

### PR DESCRIPTION
**Pull request by wasade**
Fixes #606

### CODE REVIEW

File: tests/test_table.py [hunk 1]
```diff
@@ -2233,15 +2233,17 @@ def bin_f(id_, x):
 
     def test_collapse_observations_by_metadata_one_to_many(self):
         """Collapse observations by arbitary metadata"""
-        dt_rich = Table(np.array([[5, 6, 7], [8, 9, 10], [11, 12, 13]]),
-                        ['1', '2', '3'], ['a', 'b', 'c'],
+        dt_rich = Table(np.array([[5, 6, 7], [8, 9, 10], [11, 12, 13],
```
Comment (ElDeveloper) [inline tests/test_table.py]: Instead of modifying the existing case, can you leave this one and add a new one to reproduce the bug?

**Comment by ElDeveloper** [comment]
Just one minor comment, can you also add a note to the ChangeLog?

**Comment by Jorge-C** [comment]
I'm afraid I don't understand the collapse function to comment on it, but the fix doesn't look weird ;)

**Comment by wasade** [comment]
This is an improvement on the existing test

**Comment by ElDeveloper** [comment]
I thought you said the test case did not catch this because the inputed table was squared, I wonder if we can have a case where the table is squared and one where it isn't.

**Comment by wasade** [comment]
There still is

**Comment by ElDeveloper** [comment]
Now I got it! Thanks :+1:

**Pull request merged**